Updated README.md (#42)
jaegertracing/helm-charts: charts/jaeger/README.md

## Introduction

This chart adds all components required to run Jaeger as described in the [jaeger-kubernetes](https://github.com/jaegertracing/jaeger-kubernetes) GitHub page for a production-like deployment. The chart default will deploy a new Cassandra cluster (using the [cassandra chart](https://github.com/kubernetes/charts/tree/master/incubator/cassandra)), but also supports using an existing Cassandra cluster, deploying a new ElasticSearch cluster (using the [elasticsearch chart](https://github.com/elastic/helm-charts/tree/master/elasticsearch)), or connecting to an existing ElasticSearch cluster. Once the storage backend is available, the chart will deploy jaeger-agent as a DaemonSet and deploy the jaeger-collector and jaeger-query components as Deployments.

## Installing the Chart

Add the Jaeger Tracing Helm repository:

```bash
helm repo add jaegertracing https://jaegertracing.github.io/helm-charts
```
### Default

To install the chart with the release name `jaeger`, run the following command:

```bash
helm install jaeger jaegertracing/jaeger
```

By default, the chart deploys the following:

>  - Cassandra statefulset

Reviewer: ```suggestion
 - Cassandra StatefulSet
```

 - Jaeger Agent DaemonSet
 - Jaeger Collector Deployment
 - Jaeger Query (UI) Deployment

![Jaeger with Default components](images/jaeger-default.png)

> > Image represents agent as a sidecar and for more info to refer to [this](https://medium.com/jaegertracing/deployment-strategies-for-the-jaeger-agent-1d6f91796d09)

Reviewer: ```suggestion
> Please note: The above image shows the agent collocated with the app container as a sidecar, rather than as a separate DaemonSet pod. For more info about these two approaches please refer to [this article by @jpkrohling](https://medium.com/jaegertracing/deployment-strategies-for-the-jaeger-agent-1d6f91796d09).
```

## Introduction

This chart adds all components required to run Jaeger as described in the [jaeger-kubernetes](https://github.com/jaegertracing/jaeger-kubernetes) GitHub page for a production-like deployment. 

The chart has following features :

 - Deploy or Use an existing Cassandra cluster (using the [cassandra chart](https://github.com/kubernetes/charts/tree/master/incubator/cassandra))

>  - Deploy or Use an existing ElasticSearch cluster (using the [elasticsearch chart](https://github.com/kubernetes/charts/tree/master/incubator/elasticsearch))

Reviewer: I think the previous reads a bit better. Only change the link to the ES chart, should be https://github.com/elastic/helm-charts/tree/master/elasticsearch

Could also change `back storage` for `storage backend`  and `standard individual deployments` for `Deployments`.

 - Deploy or Use an existing Kafka Cluster
 - Deploy `jaeger-agent` as a DaemonSet
 - Deploy `jaeger-collector`, `jaeger-query` and `jaeger-ingester`(optional) components as standard individual deployments.

## Installing the Chart

Add the Jaeger Tracing Helm repository:

```bash
helm repo add jaegertracing https://jaegertracing.github.io/helm-charts
```
### Default

To install the chart with the release name `jaeger`, run the following command:

```bash
helm install jaeger jaegertracing/jaeger
```

> By default, it will deploy below things in your cluster

Reviewer: ```suggestion
By default, the chart deploys the following:
```

>  - 3 node Cassandra instance

Reviewer: ```suggestion
 - Cassandra statefulset
```

kubectl apply -f jaeger-tls-cassandra-secret.yaml
helm install jaeger jaegertracing/jaeger --values values.yaml
```
### Elasticsearch

You can choose to use Elasticsearch over cassandra

**New cluster**

To install the chart with the release name `jaeger` using a new Elasticsearch cluster instead of Cassandra (default), run the following command:

```bash
helm install jaeger jaegertracing/jaeger \
  --set provisionDataStore.cassandra=false \
  --set provisionDataStore.elasticsearch=true \
  --set storage.type=elasticsearch
```

**Using an existing cluster**

A release can be configured as follows to use an existing Elasticsearch cluster as it as the storage backend:

```bash
helm install jaeger jaegertracing/jaeger \
  --set provisionDataStore.cassandra=false \
  --set storage.type=elasticsearch \
  --set storage.elasticsearch.host=<HOST> \
  --set storage.elasticsearch.port=<PORT> \
  --set storage.elasticsearch.user=<USER> \
  --set storage.elasticsearch.password=<password>
```

**Using an existing cluster with TLS**

> If you already have an existing running Elasticsearch cluster with TLS, you can configure the chart as follows to use it as your backing store:

Reviewer: Please rever this.

helm install jaeger jaegertracing/jaeger --values jaeger-values.yaml
```

### Ingester and Kafka

There are additional components which can also be installed with variety of choide for Kafka similar to Cassandra and ElasticSearch

![Jaeger with Ingester](images/jaeger-with-ingester.png)

> > Image represents agent as a sidecar and for more info to refer to [this](https://medium.com/jaegertracing/deployment-strategies-for-the-jaeger-agent-1d6f91796d09)

Reviewer: ```suggestion
```

My mistake @arpitjindal97 I took a closer look at the image, and it clearly says "host or container" meaning it represents both daemonset pod or sidecar

```bash
$ helm install jaeger jaegertracing/jaeger \
  --set provisionDataStore.kafka=true \
  --set ingester.enabled=true
```

**Using an existing Kafka cluster**

You can use an exisiting Kafak cluster with jaeger too

```bash
$ helm install jaeger jaegertracing/jaeger \
  --set provisionDataStore.kafka=false \
  --set ingester.enabled=true \
  --set storage.kafka.broker={<BROKER1:PORT>,<BROKER2:PORT>} \
  --set storage.kafka.topic=<TOPIC>
```

## Examples

### Chart with New Cassandra, Ingester and New Kafka

```bash
$ helm install jaeger jaegertracing/jaeger \
  --set provisionDataStore.kafka=true \
  --set ingester.enabled=true
```

### Chart with Existing Cassandra, Ingester and Existing Kafka

```bash
$ helm install jaeger jaegertracing/jaeger \
  --set ingester.enabled=true \
  --set provisionDataStore.cassandra=false \
  --set storage.cassandra.host=<HOST> \
  --set storage.cassandra.port=<PORT> \
  --set storage.cassandra.user=<USER> \
  --set storage.cassandra.password=<PASSWORD> \
  --set storage.kafka.broker={<BROKER1:PORT>,<BROKER2:PORT>} \
  --set storage.kafka.topic=<TOPIC>
```

### Chart with Existing Cassandra, Ingester and New Kafka

```bash

> $ helm install jaeger jaegertracing/jaeger \

Reviewer: ```suggestion
helm install jaeger jaegertracing/jaeger \
```

`$` is not needed if no ouput is shown as per https://github.com/DavidAnson/markdownlint/blob/v0.18.0/doc/Rules.md#md014

helm install jaeger jaegertracing/jaeger --values jaeger-values.yaml
```

### Ingester and Kafka

There are additional components which can also be installed with variety of choide for Kafka similar to Cassandra and Elasticsearch

> ![Jaeger with Ingester](jaeger-with-ingester.png)

Reviewer: Can we move both images into an images folder?
